### DISCUSSION TITLE
Keep running gcc9 in CI?

### DISCUSSION THREAD

**Discussion by dellaert** [borglab/gtsam#1822] [General]
One of the tests in gcc9 takes hours now (some discrete lookup test, only with TBB). I could go check out what is going on, but is gcc9 still of use? Can we just kill *all* gcc9 and save some CO2? Esp. Interested in @jlblancoc and @berndpfrommer knowledge around packaging for Linux and ROS.

**Comment by berndpfrommer** [comment]
We are currently packaging for these platforms:

Ubuntu 18.04 (bionic): gcc 7.4.0
Ubuntu 20.04 (focal): gcc 9.3.7
Ubuntu 22.04 (jammy): gcc 11.4.0

And I will try to get Ubuntu 24.04 packaging to work as well:

Ubuntu 24.04 (noble): gcc 13.2.0

It is obviously a judgement call at what point you want to drop support for older compiler versions.
Ubuntu 18.04 is EOL and in my opinion we should stop packaging for it.
Ubuntu 20.04 will go EOL in April 2025 and I think there are plenty of older robots that are still running on 20.04, in particular then ones using NVidia hardware because several of the old Jetson models were not supported beyond Ubuntu 20.04. My vote would be to keep supporting gcc 9.x until April next year.
Now obviously the old GTSAM packages will continue to be available, so it's not the end of the world either if we nix support for gcc 9.

**Comment by varunagrawal** [comment]
I think we should only consider dropping the gcc9+TBB combo for the python wrapper. That way we can still support gcc9 for the various packages, but compiling for the wrapper with TBB support is left to the user.

**Comment by dellaert** [comment]
Ok. @varunagrawal do you know how to just disable that combo? I’m good with that plan.

**Reply by varunagrawal** [reply]
All done in https://github.com/borglab/gtsam/pull/1825
